vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/WebElementsSteps.java

```diff
@@ -197,6 +203,59 @@ public void isFrameWithCertainAttributeFound(State state, String attributeType,
         baseValidations.assertElementState("The found frame is " + state, state, element);
     }
 
+    /**
+     * Gets a list of elements by <b>locator</b> and checks that it is
+     * sorted by text with using <b>sortingOrder</b>.
+     * The list should have more than 1 element. If some elements don't
+     * contain text it will be broken.
+     * @param locator locator to locate elements
+     * @param sortingOrder sorting order<br>
+     * <i>Available rule:</i>
+     * <ul>
+     * <li><b>ASCENDING</b> - will check that the list is sorted alphabetically,
+     * <li><b>DESCENDING</b> - will check that the list is sorted reverse alphabetically
+     * </ul>
+     */
+    @Then("elements located `$locator` are sorted by text in $sortingOrder order")
+    public void areElementSorted(Locator locator, SortingOrder sortingOrder)
+    {
+        List<WebElement> elements = searchActions.findElements(locator);
+        if (softAssert.assertTrue(String.format("The elements by locator: %s", locator), elements.size() >= 2))
+        {
+            List<String> textsOfElements = elements.stream()
+                    .map(webElementActions::getElementText)
+                    .collect(Collectors.toList());
+            if (softAssert.assertNotNull("The texts of elements was found", textsOfElements))
+            {
+                List<String> textsOfElementsForVerified = textsOfElements;
+                List<Integer> numberEmptyElements = new ArrayList<>();
+                for (int i = 0; i < textsOfElements.size(); i++)
+                {
+                    if (!StringUtils.isNotEmpty(textsOfElements.get(i)))
+                    {
+                        textsOfElementsForVerified.remove(i);
+                        numberEmptyElements.add(i);
+                    }
+                }
+                List<String> sortedElements = textsOfElementsForVerified.stream()
+                        .sorted((Comparator<String>) sortingOrder.getSortingType())
+                        .collect(Collectors.toList());
+                softAssert.assertEquals(String.format("The elements are sorted in %s order", sortingOrder),
+                        sortedElements, textsOfElementsForVerified);
+                if (!numberEmptyElements.isEmpty())
+                {
+                    String allNumberEmptyElements = "";
+                    for (Integer numberEmptyElement : numberEmptyElements)
+                    {
+                        allNumberEmptyElements = allNumberEmptyElements + (numberEmptyElement + 1) + " ";
+                    }
+                    throw new InternalServiceException(String.format("The element number %s"
```

Comment:
`InternalServiceException` - could you use Illegal*Exception?

Reply:
Is it expected to be thrown after order verification of "not-empty" elements? Why not throw it eagerly?

Reply:
Since we have a locator, we could use something:
Locator: //a
Element located by: (//a)[1] - has an empty text

Reply:
Or we could even calculate CSS selector using `org.vividus.ui.web.action.CssSelectorFactory`